src/aics-image-viewer/components/App/index.tsx

```diff
@@ -1159,6 +1167,48 @@ export default class App extends React.Component<AppProps, AppState> {
     return { x: 0, y: 0, z: 0 };
   }
 
+  private metadataSelectors: MetadataSelectors = {
+    dimensions: (image) => {
+      const { x, y, z } = image;
+      return { Dimensions: { x: x + "px", y: y + "px", z: z + "px" } };
+    },
+    originalDimensions: (image) => {
+      const { width, height, tiles } = image.imageInfo;
+      return { "Original dimensions": { x: width + "px", y: height + "px", z: tiles + "px" } };
+    },
+    physicalDimensions: (image) => {
+      const { physicalScale, physicalUnitSymbol, normalizedPhysicalSize } = image;
+      const [x, y, z] = normalizedPhysicalSize.toArray().map((dim: number) => dim * physicalScale + physicalUnitSymbol);
+      return { "Physical size": { x, y, z } };
+    },
+    pixelPhysicalSize: (image) => {
+      const { pixel_size, physicalUnitSymbol } = image;
+      const [x, y, z] = pixel_size.map((dim) => dim + physicalUnitSymbol);
+      return { "Physical size per pixel": { x, y, z } };
+    },
+    channels: (image) => ({ Channels: image.num_channels }),
+    timeSeriesFrames: (_image) => ({ "Time series frames": 1 }), // TODO
+    userData: (image) => image.imageInfo.userData as MetadataRecord,
+  };
+
+  getMetadata(): MetadataRecord {
+    const { metadata, metadataConfig } = this.props;
+    const { image } = this.state;
+
+    let customMetadata = {};
+    if (image && metadataConfig && metadataConfig.length > 0) {
+      metadataConfig.forEach((category) => {
+        if (category in this.metadataSelectors) {
+          const newMetadata = this.metadataSelectors[category](image);
+          customMetadata = { ...customMetadata, ...newMetadata };
+        }
```

Comment:
Just a thought: my overall preference is that this layer be a bit more "dumb" and to push these details all down into the Volume and VolumeLoaders.  Then you'd just have a Volume.getDisplayMetadata which returns all of it.  The categories could still be standardized down at that layer.  And I suppose they could still be filtered up at this level, although I'm not sure what the use case is for showing some categories and not others.

Reply:
Something like an additional field on `Volume`, set by `VolumeLoader` with whatever is available from the given format, accessible by a getter?

I guess my thought behind the category filtering was for dissemination applications like CFE, which might want to curate their metadata a bit rather than potentially overwhelming their users. But I could go either way on whether the option is useful even in that case. I was thinking of allowing clients to provide their own metadata formatting function to allow some ability to rename long property names for display (e.g. `structureSegmentationMethod`), which would also allow custom filtering. But I would be fine going with a "you get what you get" approach.